### DISCUSSION TITLE
upnp无法正常工作

### DISCUSSION THREAD

**Discussion by QZAiXH** [immortalwrt/immortalwrt#1279] [Q&A]
immortalwrt 版本：23.05.2
miniupnpd-nftables 版本：2.3.3-2

我遇到的问题与此处的[链接](https://github.com/openwrt/packages/issues/21841#issue-1854179981)描述的相同。我遵循了作者的建议进行设置，但发现执行`nft list chain inet fw4 upnp_forward`命令时返回结果为空。这导致我尝试运行`nft insert rule inet fw4 input jump upnp_forward`时失败，并且`miniupnpd`仍无法正常工作。

```
root@ImmortalWrt:~# nft list chain inet fw4 upnp_forward
table inet fw4 {
	chain upnp_forward {
	}
}
```

**Comment by chshouyu** [comment]
因为这个问题，我这几天尝试过很多渠道的固件，发现都一样
猜测可能是 luci-app-upnp 和 miniupnpd-iptables/miniupnpd-nftables 有点问题，当然也可能是固件的问题
这些渠道的固件有个共同点就是最新编译的😅
所以如果想让 upnp 正常工作，目前只能回退使用旧版本
而且我之前可能没注意，3 月到目前的各个固件都有这个问题，我回退到 2 月初的版本才恢复正常

**Reply by QZAiXH** [reply]
你说的二月初是哪个版本？回退固件还是miniupnp

**Reply by chshouyu** [reply]
回退固件，我试的这里 https://dl.openwrt.ai/firmware/x86_64/ 02.01 日期的固件是正常的

**Reply by chshouyu** [reply]
我觉得不急的话呢，可以先等等，也许过几天新固件就修复了

**Reply by yishion819** [reply]
> 回退固件，我试的这里 https://dl.openwrt.ai/firmware/x86_64/ 02.01 日期的固件是正常的

02.01以后的固件试过upnp正常么？

**Reply by chshouyu** [reply]
不正常

**Reply by leslieqwq** [reply]
> 因为这个问题，我这几天尝试过很多渠道的固件，发现都一样 猜测可能是 luci-app-upnp 和 miniupnpd-iptables/miniupnpd-nftables 有点问题，当然也可能是固件的问题 这些渠道的固件有个共同点就是最新编译的😅 所以如果想让 upnp 正常工作，目前只能回退使用旧版本 而且我之前可能没注意，3 月到目前的各个固件都有这个问题，我回退到 2 月初的版本才恢复正常

我的也这样，你们的现在解决了吗？

**Reply by ER-EPR** [reply]
在线编译的话，怎回退版本呀

**Comment by chshouyu** [comment]
https://github.com/kiddin9/OpenWrt_x86-r2s-r4s-r5s-N1/issues/1722
https://github.com/kiddin9/OpenWrt_x86-r2s-r4s-r5s-N1/issues/1711

看起来是普遍问题

**Comment by ER-EPR** [comment]
有解决了的大神么？

**Comment by hubgha163** [comment]
同样的问题
